jgrapht-core/src/main/java/org/jgrapht/alg/matching/DefaultMatching.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (C) Copyright 2016-2016, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.matching;
+
+import java.io.Serializable;
+import java.util.Set;
+
+import org.jgrapht.alg.interfaces.MatchingAlgorithm.Matching;
+
+/**
+ * A default implementation of the matching interface.
+ * 
+ * @param <E> the graph edge type
+ *
+ * @author Dimitrios Michail
+ */
+class DefaultMatching<E>
```

Comment:
In the MaximumFlowAlgorithm interface, @jsichi  and Alexey added both the MaximumFlow interface as well as a MaximumFlowImpl which implements the MaximumFlow interface. Here there are some consistency issues:

1. Naming: DefaultMatching vs MatchingImpl (same as MaximumFlowImpl)
2. Here the DefaultMatching is a dedicated class, whereas in the Flow case, the implementation is located in the MaximumFlowAlgorithm interface.

I would suggest to keep everything as consistent as possible. Let's choose a standard name, e.g DefaultMatching or MatchingImpl and enforce the same for the other interfaces. Similarly, should the default implementation reside in the interface, or in a separate class? @jsichi preferences?